src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -59,12 +63,21 @@ void RecursiveScanDirectoryTask::run() {
                 possibleCovers.append(currentFileInfo);
             }
         } else {
-            // File is a directory. Add it to our list of directories to scan.
+            // File is a directory
+            if (m_scannerGlobal->directoryBlacklisted(currentFile)) {
+                // Skip blacklisted directories
+                continue;
+            }
+            const QDir currentDir(currentFile);
+            const QString canonicalDirPath(currentDir.canonicalPath());
+            if (m_scannerGlobal->isDirectoryScanned(canonicalDirPath)) {
+                // Skip directories that have already been scanned
+                continue;
+            }
+            // Add unvisited directories to our list of directories to scan.
+            dirsToScan.append(currentDir);
             // Skip the iTunes Album Art Folder since it is probably a waste of
```

Comment:
this comment should be deleted or rephrased and moved above to the Blacklisted check